common/test/acceptance/pages/lms/problem.py

```diff
@@ -108,32 +110,80 @@ def fill_answer_numerical(self, text):
         self.wait_for_element_invisibility('.loading', 'wait for loading icon to disappear')
         self.wait_for_ajax()
 
-    def click_check(self):
+    def click_submit(self):
         """
-        Click the Check button.
+        Click the Submit button.
         """
-        self.q(css='div.problem button.check').click()
-        self.wait_for_ajax()
+        click_css(self, '.problem .submit', require_notification=False)
 
     def click_save(self):
         """
         Click the Save button.
         """
-        self.q(css='div.problem button.save').click()
-        self.wait_for_ajax()
+        click_css(self, '.problem .save', require_notification=False)
 
     def click_reset(self):
         """
         Click the Reset button.
         """
-        self.q(css='div.problem button.reset').click()
-        self.wait_for_ajax()
+        click_css(self, '.problem .reset', require_notification=False)
 
-    def click_show_hide_button(self):
-        """ Click the Show/Hide button. """
-        self.q(css='div.problem div.action .show').click()
+    def click_show(self):
+        """
+        Click the Show Answer button.
+        """
+        self.q(css='.problem .show').click()
         self.wait_for_ajax()
 
+    def is_save_notification_visible(self):
+        """
+        Is the Save Notification Visible?
+        """
+        return self.q(css='.notification.warning.notification-save').visible
+
+    def wait_for_save_notification(self):
+        """
+        Wait for the Save Notification to be present
+        """
+        self.wait_for_element_visibility('.notification.warning.notification-save',
+                                         'Waiting for Save notification to be visible')
+        self.wait_for(lambda: self.q(css='.notification.warning.notification-save').focused,
+                      'Waiting for the focus to be on the save notification')
+
+    def is_reset_button_present(self):
+        """ Check for the presence of the reset button. """
+        return self.q(css='.problem .reset').present
+
+    def is_save_button_enabled(self):
+        """ Check for the visibility of the Save button """
+        return not self.q(css='.action .save').attrs('disabled') is None
```

Comment:
@staubina This method isn't doing what you think it is (I discovered this when basing another method off of this code).

When the button has no "disabled" attributes, self.q(css='.action .save').attrs('disabled') returns a list [None]. That isn't None, but it has nothing to do with whether or not the button is disabled.... When the button IS disabled, you get an array with other values ('true', I believe).

My point is that this method will always return True, regardless of the disabled state. Can you please fix it and investigate the other method(s) that check the disabled attribute?